components/jira/jira.app.mjs

```diff
@@ -1,11 +1,257 @@
+import { axios } from "@pipedream/platform";
+
 export default {
   type: "app",
   app: "jira",
-  propDefinitions: {},
+  propDefinitions: {
+    projectID: {
+      type: "string",
+      label: "Project ID",
+      description: "The project ID.",
+      async options() {
+        const resp = await this._makeRequest({
+          method: "GET",
+          path: "/project",
+        });
+        return resp.map((e) => ({
+          label: e.name,
+          value: e.id,
+        }));
+      },
+    },
+    issueType: {
+      type: "string",
+      label: "Issue Type",
+      description: "An ID identifying the type of issue, [Check the API docs](https://developer.atlassian.com/cloud/jira/platform/rest/v3/#api-rest-api-3-issue-post) to see available options",
+      async options({ projectID }) {
+        const resp = await this._makeRequest({
+          method: "GET",
+          path: "/issue/createmeta",
+          params: {
+            projectIds: projectID,
+          },
+        });
+        // Because we select a project, we get issue types for only that and only one project
+        return resp.projects[0].issuetypes.map((e) => ({
+          label: e.name,
+          value: e.id,
+        }));
+      },
+    },
+    issueIdOrKey: {
+      type: "string",
+      label: "Issue id or key",
+      description: "The ID or key of the issue where the attachment will be added to.",
+    },
```

Comment:
Can we use async options for `issueIdOrKey` to list the available issues?
https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-issue-search/#api-rest-api-3-search-get

Reply:
I think there can be a lot of issues and user cannot find by selecting from list and loading more and more. And when fetching them, they come with some extra data, which can make it huge. This usage may not be very handful. What do you think? Please correct me if I am wrong